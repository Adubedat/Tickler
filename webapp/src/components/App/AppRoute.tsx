import { Redirect, Route, RouteProps } from 'react-router-dom';

type Props = {
    isAuth: Boolean;
} & RouteProps;

export const PublicRoute = ({ isAuth, ...props }: Props) => {
    return isAuth
        ? (<Redirect to="/dashboard" />)
        : (<Route {...props} />)
};

export const ProtectedRoute =  ({ isAuth, ...props }: Props) => {
    return isAuth
        ? (<Route {...props} />)
        : (<Redirect to="/login" />)
};