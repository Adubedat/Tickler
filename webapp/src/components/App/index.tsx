 
import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
} from 'react-router-dom';
import routes from '../../config/routes';
import { BackgroundContainer } from './styles';
import { UserProvider } from '../../context/User';
 
function App() {
  return (
    <UserProvider>
      <BackgroundContainer>
        <Router>
          <Switch>
            {routes.map((route) => (
              <Route
                key={route.path}
                exact path={route.path}
              >
                <route.component />
              </Route>
              ))}
          </Switch>
        </Router>
      </BackgroundContainer>
    </UserProvider>
  );
}
 
export default App;
