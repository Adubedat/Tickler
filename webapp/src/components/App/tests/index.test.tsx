import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import App from "../index";

describe("<App />", () => {
  test("renders", () => {
    const wrapper = shallow(<App />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});