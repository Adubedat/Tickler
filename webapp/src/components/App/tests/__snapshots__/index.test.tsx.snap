// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<styled.div>
  <BrowserRouter>
    <Switch>
      <PublicRoute
        component={[Function]}
        exact={true}
        isAuth={false}
        path="/"
      />
      <PublicRoute
        component={[Function]}
        exact={true}
        isAuth={false}
        path="/register"
      />
      <ProtectedRoute
        component={[Function]}
        exact={true}
        isAuth={false}
        path="/dashboard"
      />
    </Switch>
  </BrowserRouter>
</styled.div>
`;
